crates/sui-types/src/messages.rs

```diff
@@ -1423,11 +1423,16 @@ impl TransactionEffects {
     /// created and unwrapped objects. In other words, all objects that still exist
     /// in the object state after this transaction.
     /// It doesn't include deleted/wrapped objects.
-    pub fn all_mutated(&self) -> impl Iterator<Item = &(ObjectRef, Owner)> + Clone {
+    pub fn all_mutated(&self) -> impl Iterator<Item = (&ObjectRef, &Owner, WriteKind)> + Clone {
```

Comment:
Does this function need to return the WriteKind?

Reply:
Yes, because of gateway_state. It pulls information out of transaction effects, modifies it, and puts it back into a new TemporaryStore, which now needs the write kind